src/content/docs/en/reference/adapter-reference.mdx

```diff
@@ -188,3 +202,124 @@ You can usually call `app.render(request)` without using `.match` because Astro
 ```
 
 Once you [publish your adapter to npm](https://docs.npmjs.com/cli/v8/commands/npm-publish), running `astro add example` will install your package with any peer dependencies specified in your `package.json`. We will also instruct users to update their project config manually.
+
+## Adapter features
+
+A set of features that changes the output of the emitted files. When an adapter opt ins these features, they will get additional
+information in specific hooks.
```

Comment:
Just a small suggestion to make the sentence a bit easier to understand (feel free to refine further if I misunderstood the sentence)

```suggestion
A set of features that changes the output of the emitted files. When an adapter opts in to these features, they will get additional information inside specific hooks.
```